### PR TITLE
Get cpu client working on freebsd

### PR DESCRIPTION
**Pull request by rminnich**
needs testing ...

### CODE REVIEW

**Comment by rminnich** [comment]
The cpu command works on freebsd now
PWD=/ ./cpu -namespace=/home -9p=false -nfs=true -sp 22 linoderoot
works

**Comment by rminnich** [comment]
I've tested it and it works, so that's something anyway

**Comment by bsdimp** [comment]
This is cool... I wonder why there's no error for the permission thing...

File: client/cpu9p_unix_test.go [hunk 1]
```diff
@@ -178,13 +179,14 @@ func Test9pUnix(t *testing.T) {
...
 	if err := c.SetAttr(sam, sa); err != nil {
-		t.Fatalf("Setattr with mode: %v != nil", err)
+		t.Fatalf("Setattr(%v,%v): %v != nil", sam, sa, err)
 	}
 
 	// The second time, since we blew permissions to
 	// zero, we ought to get an error.
-	if err := c.SetAttr(sam, sa); err == nil {
-		t.Fatalf("Setattr with mode: nil != %v", unix.EPERM)
+	// It seems, on freebsd, there is no error.
```
Comment (bsdimp) [inline client/cpu9p_unix_test.go:187]: I wonder why that's the case.

File: client/cpu_freebsd.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2018-2019 the u-root Authors. All rights reserved
```
Comment (bsdimp) [inline client/cpu_freebsd.go]: you just wrote this, no?? :)

File: client/getattr_freebsd.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 The gVisor Authors.
```
Comment (bsdimp) [inline client/getattr_freebsd.go:1]: Similar?

File: client/getattr_freebsd.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+import (
+	"syscall"
+
+	"github.com/hugelgupf/p9/p9"
+)
+
+// GetAttr implements p9.File.GetAttr.
+//
+// Not fully implemented.
```
Comment (bsdimp) [inline client/getattr_freebsd.go]: Is it easy to articulate what's missing?

**Review by bsdimp — APPROVED** [review]
If it works, it works :)

There were a couple of questions I had. I can almost read 'go' most days, but this looks good from what I can tell.

File: client/cpu9p_unix.go [hunk 5]
```diff
@@ -34,7 +35,7 @@ func (l *CPU9P) SetAttr(mask p9.SetAttrMask, attr p9.SetAttr) error {
 
 	if mask.Size {
 		if e := unix.Truncate(l.path, int64(attr.Size)); e != nil {
-			err = errors.Join(err, e)
+			err = errors.Join(err, fmt.Errorf("Truncate:%w", err))
```
Comment (binjip978) [inline client/cpu9p_unix.go]: nit: `fmt.Errorf("truncate:%w", err)`

**Review by binjip978 — APPROVED** [review]

File: client/cpu9p_unix_test.go (hunk 1 above)
Comment (rminnich) [reply client/cpu9p_unix_test.go:187]: I am hoping you can tell me :-)

File: client/cpu_freebsd.go (hunk 2 above)
Comment (rminnich) [reply client/cpu_freebsd.go]: fixed.

File: client/getattr_freebsd.go (hunk 3 above)
Comment (rminnich) [reply client/getattr_freebsd.go:1]: it's originally a gvisor thing so this one I want to leave as is. It's not much code.

File: client/getattr_freebsd.go (hunk 4 above)
Comment (rminnich) [reply client/getattr_freebsd.go]: I think, looking at it, that it's actually complete now.

File: client/cpu9p_unix.go (hunk 5 above)
Comment (rminnich) [reply client/cpu9p_unix.go]: fixed.

**Pull request merged**